### DISCUSSION TITLE
Measuring contributor growth over time, and monthly active contributors

### DISCUSSION THREAD

**Discussion by sideshowbarker** [mdn/mdn-community#485] [Content]
I wrote [a tool for generating historical monthly committer totals for any given git repo](https://github.com/sideshowbarker/git-monthly-committer-totals/). Current results for https://github.com/mdn/content:

```
      Committers
┌────────────┬───────┐
│   Monthly  │ Cu-   │
├──────┬─────┤ mula- │ Monthly
│ Uniq │ New │ tive  │ commits             Month range
├──────┼─────┼───────┼───────┬──────────────────────────────────────────────┐
│  197 │ 142 │  1332 │   434 │ Oct 20 – Nov 19 2021 (ending 5 months ago)   │
│  155 │ 106 │  1438 │   380 │ Nov 20 – Dec 19 2021 (ending 4 months ago)   │
│  200 │ 141 │  1579 │   513 │ Dec 20 – Jan 19 2022 (ending 3 months ago)   │
│  194 │ 138 │  1717 │   564 │ Jan 20 – Feb 19 2022 (ending 2 months ago)   │
│  194 │ 135 │  1852 │   498 │ Feb 20 – Mar 19 2022 (ending 1 month ago)    │
│  183 │ 126 │  1978 │   684 │ Mar 20 – Apr 19 2022 (ending today)          │
└──────┴─────┴───────┴───────┴──────────────────────────────────────────────┘
 For the last 6 months:
     512 commits per month on average.
     187 unique committers per month on average.
     131 new committers per month on average.
   Cumulative total committers grew by 646 (from 1332 to 1978) in 5 months.
```
You can try the tool yourself but running the following within your https://github.com/mdn/content clone —

```
bash -c \
  "$(curl -fsSL https://sideshowbarker.github.io/git-monthly-committer-totals/git-totals.sh)" "" 6
```
Change the `6` to whatever number of months of history you want to look at.

---

I also recently came across https://github.com/api7/contributor-graph, which is a tool for generating graphs of contributor growth over time for one or more GitHub repos — as well as graphing the number of monthly active contributors.

The graphs for the mdn/content repo are here:

- https://git-contributor.com/?chart=contributorOverTime&repo=mdn/content
- https://git-contributor.com/?chart=contributorMonthlyActivity&repo=mdn/content

![Contributor growth over time](https://contributor-overtime-api.apiseven.com/contributors-svg?chart=contributorOverTime&repo=mdn/content)

![Monthly active conributors](https://contributor-overtime-api.apiseven.com/contributors-svg?chart=contributorMonthlyActivity&repo=mdn/content)

**Comment by mturoci** [comment]
https://sideshowbarker.github.io/git-monthly-committer-totals/git-totals.sh throws 404 unfortunately.
